test/functional/legacy/031_close_commands_spec.lua

```diff
@@ -84,6 +84,29 @@ describe('Commands that close windows and/or buffers', function()
     feed('GA 4<Esc>:all!<CR>')
     execute('1wincmd w')
     expect('testtext 2 2 2')
+
+    -- Test ":q!" and hidden buffer.
+    execute('bw! Xtest1 Xtest2 Xtest3 Xtest4')
+    execute('sp Xtest1')
+    execute('wincmd w')
+    execute('bw!')
+    execute('set modified')
+    execute('bot sp Xtest2')
+    execute('set modified')
+    execute('bot sp Xtest3')
+    execute('set modified')
+    execute('wincmd t')
+    execute('hide')
+    execute('q!')
+    expect('testtext 3')
+    execute('q!')
+    feed('<CR>')
+    expect('testtext 1')
+    execute('q!')
+    -- Now nvim has exited
+    assert.has_error(function()
+      execute('call append(line("$"), "Oh, Not finished yet.")')
+    end)
```

Comment:
To workaround the segfault you could put the code in a single `source([[...]])` call, and that should be equivalent to the old test?